src/vg_active_segment.erl

```diff
@@ -41,12 +41,33 @@ start_link(Topic, Partition, NextBrick) ->
     gen_server:start_link(?SERVER(Topic, Partition), ?MODULE, [Topic, Partition, NextBrick, Tab], []).
 
 write(Topic, Partition, MessageSet) ->
-    gen_server:call(?SERVER(Topic, Partition), {write, MessageSet}).
+    %% there clearly needs to be a lot more logic here.  it's also not
+    %% clear that this is the right place for this
+    try
+        gen_server:call(?SERVER(Topic, Partition), {write, MessageSet})
+    catch C:E ->
```

Comment:
this needs to be refined in two ways:

1. it needs to be configurable, since we won't always want to create new topics when written to
2. it needs to be more precise, as not all errors here are going to mean that it isn't running.